### PR TITLE
[4.0] Removing eslint from hound and adding to drone CI

### PR DESCRIPTION
**Pull request by Hackwar**
Pull Request for Issue #28100.

### Summary of Changes
This moves eslint from hound to drone CI, thus using the eslint from our package.json.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -65,6 +65,12 @@ steps:
       branch:
       - 4.0-dev
 
+  - name: eslint
+    image: node:current-alpine
+    depends_on: [ npm ]
+    commands:
+      - ./node_modules/.bin/eslint -c .eslintrc --ignore-path .eslintignore **/*.js
```
Comment (HLeithner) [inline .drone.yml]:
```suggestion
      - ./node_modules/.bin/eslint -c .eslintrc --ignore-path .eslintignore "**/*.js"
```

**Comment by wilsonge** [comment]
I'm tempted to stick with hound. Only because it comments with suggestions for how to fix - which to me feels like it's going to be easier for new contributors or those with less technical experience?

File: .drone.yml [hunk 2]
```diff
@@ -65,6 +65,12 @@ steps:
       branch:
       - 4.0-dev
 
+  - name: eslint
+    image: node:current-alpine
+    depends_on: [ npm ]
+    commands:
+      - ./node_modules/.bin/eslint -c .eslintrc --ignore-path .eslintignore "**/*.js"
```
Comment (dgrammatiko) [inline .drone.yml:72]: > `./node_modules/.bin/eslint -c .eslintrc --ignore-path .eslintignore "**/*.js"`

can also be written as:
` npx eslint -c .eslintrc --ignore-path .eslintignore "**/*.js"`

**Comment by dgrammatiko** [comment]
> I'm tempted to stick with hound

Please reconsider. Hound was a service that fulfilled something that wasn't possible couple of years ago in GitHub. Nowadays we have actions, so a simple script can take all the artefacts from each PR (PHPCS. ESLINT, SCSS LINT. etc) and expose them in the PR as a comment or comments. So whatever Hound is doing is totally within the reach of plain GitHub Actions and without having to sync different services (it's a mess). Eg since this was kicked for ESlint: https://github.com/marketplace/actions/eslint-action

PS. Not to mention that you can offload **all** the drone work to Github action and reduce the bills...

**Comment by brianteeman** [comment]
I agree with @wilsonge about the way that hound comments being a good thing. also drone fails so often its not a useful tool at all

**Comment by dgrammatiko** [comment]
It won’t be drone but Github actions if for once you hear what I’m saying...

**Comment by brianteeman** [comment]
If you read more carefully you would have realised I meant the way that drone comments is good. I dont care what tool is used as long as it actually works. (drone doesnt)

**Comment by wilsonge** [comment]
Ok Eslint GitHub actions seems viable from a quick glance in terms of the comments it gives. However wouldn’t we then have to maintain + support it?

**Comment by dgrammatiko** [comment]
> However wouldn’t we then have to maintain + support it?

Let's be realistic here, right now you admin and maintain (and also **pay**) for drone and travis. Moving these tasks to GH Actions will remove the administrative work (server pampering) but of course will not eliminate casual maintenance, it's still executable code (although yml) so same rules apply. What you'll remove from the equation are the own servers, which should be beneficial long term (less people doing plumping instead of contributing actual code for the product, etc).
FWIW I think it will be better if the production team sets some goals for moving to GH actions and start moving small chunks. Eg all the js related parts: lint, tests etc can be moved out of DRONE right now.
The only part that needs some consideration and probably figure out the optimal solution is that the actions, if implemented right now, will target the branch `4.0-dev` but once you'll switch to staging there will be a need for some adjustments. Also unfortunately you **can't** play with the actions in your local env right now so also another negative...


PS: `you` as `the project`

**Comment by dgrammatiko** [comment]
@wilsonge @Hackwar it might worth the time investigating moving 
- js lint: https://github.com/marketplace/actions/eslint-action
- phpcs: https://github.com/chekalsky/phpcs-action
- scss lint: https://github.com/marketplace/actions/stylelinter
to GitHub Actions. This is really low hanging fruit and will improve the DX on the branch

Such implementation will leave only the tests on the DRONE machine so, theoretically, should be faster to get the results for the codestyle errors

**Comment by C-Lodder** [comment]
@wilsonge I use GH Actions on some of my own personal projects. You simply tell it to install NodeJS, run `npm i` and then whatever tasks you wish to run. There is no maintenance required and it's much better keeping things in-house when possible. It took 2 minutes to setup when testing the WebAuth PR

**Comment by wilsonge** [comment]
Closing this. As is it isn't the correct tool. At minimum to move to drone you need to add the ability to annotate PRs into this. But would be more than happy to see some work moving to actions if we wanted to go that way too

**Pull request closed without merging**